sdk/storage/azure-storage-files-shares/src/share_lease_client.cpp

```diff
@@ -62,50 +65,41 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
     }
     else
     {
-      std::abort();
+      AZURE_UNREACHABLE_CODE;
     }
   }
 
   Azure::Response<Models::RenewLeaseResult> ShareLeaseClient::Renew(
-      const RenewLeaseOptions& options,
+      const RenewLeaseOptions&,
       const Azure::Core::Context& context) const
   {
-    (void)options;
-    if (m_fileClient.HasValue())
-    {
-      // Renew only support share level lease.
-      std::abort();
-    }
-    else if (m_shareClient.HasValue())
-    {
-      _detail::ShareRestClient::Share::RenewLeaseOptions protocolLayerOptions;
-      protocolLayerOptions.LeaseIdRequired = m_leaseId;
+    AZURE_ASSERT(!m_fileClient.HasValue());
+    AZURE_ASSERT(m_shareClient.HasValue());
```

Comment:
what if we support file-level renew in the future?

Do not change the logic in this PR. Do it in a separate PR

Reply:
> what if we support file-level renew in the future?

We would change/remove the precondition just the same as we would remove the current abort() call from the implementation. It is the same exact thing man! :)

Reply:
why not just replace the `std::abort` with `AZURE_UNREACHABLE_CODE`?

Reply:
If file-level renew is supported in the future, we will have to add the `if {} else if {} else {}` back.